crates/sui-data-ingestion/src/executor.rs

```diff
@@ -22,15 +22,15 @@ pub struct IndexerExecutor<P> {
     pools: Vec<Pin<Box<dyn Future<Output = ()> + Send>>>,
     pool_senders: Vec<mpsc::Sender<CheckpointData>>,
     progress_store: ProgressStoreWrapper<P>,
-    pool_progress_sender: mpsc::Sender<(String, CheckpointSequenceNumber)>,
-    pool_progress_receiver: mpsc::Receiver<(String, CheckpointSequenceNumber)>,
+    pool_progress_sender: mpsc::UnboundedSender<(String, CheckpointSequenceNumber)>,
+    pool_progress_receiver: mpsc::UnboundedReceiver<(String, CheckpointSequenceNumber)>,
     metrics: DataIngestionMetrics,
 }
 
 impl<P: ProgressStore> IndexerExecutor<P> {
+    #[allow(clippy::disallowed_methods)] // channel is logically bounded by number_of_worker_jobs * MAX_CHECKPOINTS_IN_PROGRESS
```

Comment:
Should we bound the channel to size `MAX_NUM_WORKER_JOBS * MAX_CHECKPOINTS_IN_PROGRESS`? Basically, suggesting to keep some theoretical upper bound (to prevent unbounded mem growth) vs an unbounded channel.

Reply:
yes, but we don't know the value in advance with the existing API.
But we can change the API